numpy/lib/shape_base.py

```diff
@@ -758,7 +759,54 @@ def array_split(ary, indices_or_sections, axis=0):
     >>> np.array_split(x, 3)
         [array([0.,  1.,  2.]), array([3.,  4.]), array([5.,  6.])]
 
+    >>> x = np.reshape(np.arange(16), (4, 4))
+    >>> np.array_split(x, [3, 3], (0, 1))
+        [array([[0, 1],
+                [4, 5]]),
+        array([[2], [6]]),
+        array([[3], [7]]),
+        array([[8, 9]]),
+        array([[10]]),
+        array([[11]]),
+        array([[12, 13]]),
+        array([[14]]),
+        array([[15]])]
+
+    >>> x = np.reshape(np.arange(8), (2, 2, 2))
+    >>> np.array_split(x, [2, 2, 2], (0, 1, 2))
+        [array([[[0]]]),
+        array([[[1]]]),
+        array([[[2]]]),
+        array([[[3]]]),
+        array([[[4]]]),
+        array([[[5]]]),
+        array([[[6]]]),
+        array([[[7]]])]
```

Comment:
Formatting is off here:
```suggestion
    >>> np.array_split(x, 3)
    [array([0., 1., 2.]), array([3., 4.]), array([5., 6.])]
    >>> x = np.reshape(np.arange(16), (4, 4))
    >>> np.array_split(x, [3, 3], (0, 1))
    [array([[0, 1],
            [4, 5]]),
     array([[2], [6]]),
     array([[3], [7]]),
     array([[8, 9]]),
     array([[10]]),
     array([[11]]),
     array([[12, 13]]),
     array([[14]]),
     array([[15]])]
    >>> x = np.reshape(np.arange(8), (2, 2, 2))
    >>> np.array_split(x, [2, 2, 2], (0, 1, 2))
    [array([[[0]]]),
     array([[[1]]]),
     array([[[2]]]),
     array([[[3]]]),
     array([[[4]]]),
     array([[[5]]]),
     array([[[6]]]),
     array([[[7]]])]
```